doto-frontend/src/components/pages/Calendar/TaskScheduler.js

```diff
@@ -23,17 +23,35 @@ const addTaskToSchedule = (newTask, existingTasks, currDate) => {
 
     // Separate existing tasks with a startDate > current datetime from older tasks (which won't be considered)
     existingTasks.forEach(task => {
-        task.startDate > currDate ? competingTasks.push(task) : oldTasks.push(task);
+        task.startDate > newTask.earliestDate ? competingTasks.push(task) : oldTasks.push(task);
     });
 
-    // If the endDate of the latest oldTask is after currDate, then the earliest any new task can be scheduled is oldTask.endDate
+    // If the endDate of the latest oldTask is after earliest start date, then the earliest any new task can be scheduled is oldTask.endDate
     const minDate =
-        (oldTasks[0] && oldTasks[oldTasks.length - 1].endDate) > currDate
+        (oldTasks[0] && oldTasks[oldTasks.length - 1].endDate) > newTask.earliestDate
             ? oldTasks[oldTasks.length - 1].endDate
-            : currDate;
+            : newTask.earliestDate;
 
     let cTask;
 
+    const earliestPossibleEnd = new Date(
+        newTask.earliestDate.getTime() +
+            newTask.duration * MILLISECONDS_PER_MINUTE +
+            newTask.travelTime * MILLISECONDS_PER_MINUTE,
+    );
+
+    if (competingTasks.length > 0) {
+        if (earliestPossibleEnd < competingTasks[0].startDate && newTask.dueDate < competingTasks[0].dueDate) {
+            newTask.startDate = new Date(newTask.earliestDate.getTime());
+            newTask.endDate = earliestPossibleEnd;
+
```

Comment:
Just noticed that `reminderDate` is not being set here.
And its not being set in the other exit condition either.
Maybe it would be a good idea to extract setting properties on `newTask` to a separate function?

Reply:
Added reminderDate being set.

> Maybe it would be a good idea to extract setting properties on newTask to a separate function?

Might be better to do that when we refactor scheduling for the back-end in that case?

Reply:
Ok sounds good